modules/serverless-ossm-v1x-tls.adoc

```diff
@@ -0,0 +1,157 @@
+// Module included in the following assemblies:
+//
+// * serverless/networking/serverless-ossm-tls.adoc
+
+[id="serverless-ossm-v1x-tls_{context}"]
+= Configuring TLS for a custom domain using {ProductName} 1.x
+
+.Procedure
+
+. Create a root certificate and private key to sign the certificates for your services:
++
+[source,terminal]
+----
+$ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 \
+    -subj '/O=Example Inc./CN=example.com' \
+    -keyout example.com.key \
+    -out example.com.crt
+----
+. Create a certificate signing request for your domain:
++
+[source,terminal]
+----
+$ openssl req -out custom.example.com.csr -newkey rsa:2048 -nodes \
+    -keyout custom.example.com.key \
+    -subj "/CN=custom-ksvc-domain.example.com/O=Example Inc."
+----
+. Sign the request with your CA:
++
+[source,terminal]
+----
+$ openssl x509 -req -days 365 -set_serial 0 \
+    -CA example.com.crt \
+    -CAkey example.com.key \
+    -in custom.example.com.csr \
+    -out custom.example.com.crt
+----
+. Check that the certificates appear in your directory:
++
+[source,terminal]
+----
+$ ls -1
+----
++
+.Example output
+[source,terminal]
+----
+custom.example.com.crt
+custom.example.com.csr
+custom.example.com.key
+example.com.crt
+example.com.key
+----
+. Create a secret:
++
+[source,terminal]
+----
+$ oc create -n istio-system secret tls custom.example.com \
+    --key=custom.example.com.key \
+    --cert=custom.example.com.crt
+----
+. Attach the secret to the Istio ingress gateway, by editing the `ServiceMeshControlPlane` resource.
+.. Edit the `ServiceMeshControlPlane` resource:
++
+[source,terminal]
+----
+$ oc edit -n istio-system ServiceMeshControlPlane <control_plane_name>
+----
+.. Check that the following lines exist in the resource, and if they do not, add them:
++
+[source,yaml]
+----
+istio-ingressgateway:
```

Comment:
We found that it is more complicated. (Thanks you @cardil )

If maistra is `maistra.io/v1` then we need to  configure the secret as current doc:

```
    gateways:
      istio-ingressgateway:
        secretVolumes:
        - mountPath: /custom.example.com
          name: custom-example-com
          secretName: custom.example.com
```

But  if maistra is `maistra.io/v2`, then we need to  configure the secret as v2 doc:

```
spec:
  gateways:
    ingress:
      volumes:
      - volume:
          secret:
            secretName: custom.example.com
        volumeMount:
          name: custom-example-com
          mountPath: /custom.example.com
```

Reply:
NOTE: `maistra.io/v1` was not used if ServiceMesh operator is installed recently but if users installed the SM operator long time ago, it still uses `maistra.io/v1`.

Reply:
Confirmed that this works.

There are 4 cases that might happen here:

1. User is holding off to old 1.x version of SM operator
1. User has upgraded to SM operator 2.x but still has SMCP defined using `maistra.io/v1` spec
1. User has installed or upgraded SM operator to 2.x, and migrated to using `maistra.io/v2` SMCP crd but with `spec.version == 1.0` or `spec.version == 1.1`
1. User has installed SM operator 2.x and it's using `maistra.io/v2` SMCP crd with default options, or with `spec.version == 2.0` set

In 1st (the least likely option) and 2nd the user should follow 1.x docs, in other cases user should follow 2.x docs.

Reply:
@abrennan89 What about stating somewhere in the header of this page that:

> Serverless is supporting only SM 1.x or 2.x. Serverless do not support all others intermidiate states that SM currently offers (ref. page [upgrading-ossm](https://docs.openshift.com/container-platform/4.6/service_mesh/v2x/upgrading-ossm.html))

After including such a statement, I'm okay with closing this thread.

Reply:
Apart from one simple thing (https://github.com/openshift/openshift-docs/pull/27660/files#r561670112), all is working for old version of SM and for new fully upgraded one.

![Zrzut ekranu z 2021-01-20 21-04-34](https://user-images.githubusercontent.com/540893/105323700-b7cb2f80-5bca-11eb-875a-7bd50df6d078.png)